supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/Postprocessor.ts

```diff
@@ -0,0 +1,430 @@
+     */
+    findMethodCallsFromMethodNameAndClassBody(methodName: string, classBody: TSESTree.ClassDeclaration, restCall: TSESTree.CallExpression) {
+        let methodCalls: { callExpression: TSESTree.CallExpression, parameterValue: string }[] = [];
+        let parsingResultType = ParsingResultType.GET_URLS_FROM_CALLING_METHODS_FAILURE;
+
+        for (const methodDefinition of classBody.body.body) {
+            if (methodDefinition.type === 'MethodDefinition' && methodDefinition.key.type === 'Identifier') {
+                simpleTraverse(methodDefinition, {
+                    enter: (node) => {
+                        if (node.type === 'CallExpression' && node.callee.type === 'MemberExpression' && node.callee.object.type === 'ThisExpression' && node.callee.property.type === 'Identifier' &&
+                            node.callee.property.name === methodName && node.arguments[0].type === 'Identifier') {
+                            let parameterValue = this.getMethodVariableValueFromNameAndMethod(node.arguments[0].name, node, methodDefinition, restCall, classBody);
+                            if (parameterValue.resultType === ParsingResultType.GET_VARIABLE_FROM_METHOD_SUCCESS) {
+                                methodCalls.push({ callExpression: node, parameterValue: parameterValue.result[0] });
+                            } else if (parameterValue.resultType === ParsingResultType.UNABLE_TO_EVALUATE) {
+                                parsingResultType = ParsingResultType.UNABLE_TO_EVALUATE;
+                            }
+                        }
+                    }
+                });
+            }
+        }
+
+        let result: string[] = [];
+        for (const methodCall of methodCalls) {
+            result.push(methodCall.parameterValue);
+        }
+
+        if (result.length > 0) {
+            parsingResultType = ParsingResultType.GET_URLS_FROM_CALLING_METHODS_SUCCESS;
+        }
+        return new ParsingResult(parsingResultType, result);
+    }
+
+    getMemberExpressionValueFromNameAndClassBody(name: string, classBody: TSESTree.ClassDeclaration) {
+        let memberExpressionResult: string[] = [];
+        let resultType = ParsingResultType.EVALUATE_MEMBER_EXPRESSION_FAILURE;
+        simpleTraverse(classBody, {
+            enter(node) {
+                if (node.type === 'PropertyDefinition' && node.value?.type === 'Literal' && node.key.type === 'Identifier' && node.value.value) {
+                    if (node.key.name === name) {
+                        memberExpressionResult.push(node.value.value.toString());
+                        resultType = ParsingResultType.EVALUATE_MEMBER_EXPRESSION_SUCCESS;
+                        return;
+                    }
+                }
+            }
+        });
+        return new ParsingResult(resultType, memberExpressionResult);
+    }
+
+    getMemberExpressionTypeFromName(name: string) {
```

Comment:
I think this returns a `string`?

```suggestion
    getMemberExpressionTypeFromName(name: string): string {
```